### PR TITLE
#167574995 Enable Social Login via Google

### PR DESCRIPTION
**Pull request by Adekoreday**
#### What does this PR do?
This PR enables the existing user to log in and new user  signup to the application via google

#### Description of Task proposed in this pull request?
- This PR uses the passportjs Google strategy to the codebase, which enables authenticating the user via google. The core implementation of this feature exists in the helper function google.js and it was integrated into our existing API for facebook authentication.

#### How should this be manually tested (Quality Assurance)?
 ```
npm run start: dev
open your browser and go to the following URL
http:locahost:5000/api/v1/auth/google
The user is redirected to the following URL automatically
http:localhost:5000/api/v1/auth/google/callback



   ```

#### What are the relevant pivotal tracker stories?
[167574995](https://www.pivotaltracker.com/story/show/167574995)

#### Any background context you want to add?
```
- npm install
-add the following .env variables
GOOGLE_CLIENT_ID=
GOOGLE_CLIENT_SECRET=
GOOGLE_REDIRECT_URL=http://localhost:5000/api/v1/auth/google/callback
```
#### What I have learned working on this feature:
- I learnt how to authenticate user using passport

#### Screenshots: 
Screenshot 2019-08-05 at 11 49 17 PM
Screenshot 2019-08-05 at 11 49 48 PM

### CODE REVIEW

**Review by JuwonAbiola — DISMISSED** [review]

**Review by chialuka — DISMISSED** [review]

**Review by JuwonAbiola — DISMISSED** [review]

**Review by henryade — DISMISSED** [review]

File: test/auth/facebook.test.js [hunk 1]
```diff
@@ -2,33 +2,39 @@ import chai from 'chai';
...
   });
 
   after(() => {
     sinons.restore();
+    deserialize.restore();
+    serialize.restore();
   });
 
-  it('it should test route nock off the http request to fb', async (done) => {
+  it('it should test route nock off the http request to fb', async () => {
```
Comment (dinobi) [inline test/auth/facebook.test.js]: `it should` pls fix this

File: test/auth/facebook.test.js [hunk 2]
```diff
@@ -2,33 +2,39 @@ import chai from 'chai';
...
     nock('https://wwww.facebook.com')
       .get('/api/v1/auth/facebook/callback')
       .reply(200, {
         hello: 'facebook'
       });
-    done();
   });
 
   it('it mocks passport app', async () => {
```
Comment (dinobi) [inline test/auth/facebook.test.js]: and this

File: test/auth/facebook.test.js [hunk 3]
```diff
@@ -63,23 +69,26 @@ describe('facebook login test', () => {
...
     const json = sinon.stub(res, 'json').returnsThis();
-    getSocialUserData(null, '');
+    getSocialUserData(mockrequest);
     PassportError.passportErrors(false, Callbackrequest, res, Callbacknext);
     sinon.assert.calledOnce(Callbacknext);
     sinon.assert.calledOnce(status);
     sinon.assert.calledOnce(json);
   });
-
   it('it facebookLoginController', async () => {
```
Comment (dinobi) [inline test/auth/facebook.test.js]: and this

File: test/auth/facebook.test.js [hunk 4]
```diff
@@ -63,23 +69,26 @@ describe('facebook login test', () => {
...
     const Sessions = sinon.stub(Session, 'create').returns(false);
     const findOrCreate = sinon.stub(User, 'findOrCreate').returns(users);
-    await Auth.facebookSocialLogin(request, res);
+    await Auth.socialLogin(request, res);
     sinon.assert.calledOnce(status);
     sinon.assert.calledOnce(json);
     Sessions.restore();
     findOrCreate.restore();
   });
+  it('it test facebook helpers', async () => {
```
Comment (dinobi) [inline test/auth/facebook.test.js]: and this

File: test/auth/google.test.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import sinon from 'sinon';
+import google from '../../server/helpers/google';
+import Auth from '../../server/controllers/Auth';
+import models from '../../server/database/models/index';
+import { request2, users } from './__mocks_';
+
+const { expect } = chai;
+const { User, Session } = models;
+describe('google login test', () => {
+  context('when user successfully login', () => {
```
Comment (dinobi) [inline test/auth/google.test.js]: the use case for `context` here doesn't make sense

File: test/auth/google.test.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    it('it  Logs in with google', async () => {
```
Comment (dinobi) [inline test/auth/google.test.js]: fix this

File: test/auth/google.test.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+      const findOrCreate = sinon.stub(User, 'findOrCreate').returns(users);
+      await Auth.socialLogin(request2, res);
+      sinon.assert.calledOnce(status);
+      sinon.assert.calledOnce(json);
+      Sessions.restore();
+      findOrCreate.restore();
+    });
+  });
+  context('test google strategy', () => {
+    it('it test google strategy', async () => {
```
Comment (dinobi) [inline test/auth/google.test.js]: fix this

**Review by dinobi — CHANGES_REQUESTED** [review]
Please address all the comments
done thanks

**Review by abdulfatai360 — DISMISSED** [review]

File: test/auth/facebook.test.js (hunk 3 above)
Comment (Adekoreday) [reply test/auth/facebook.test.js]: done

File: test/auth/facebook.test.js (hunk 4 above)
Comment (Adekoreday) [reply test/auth/facebook.test.js]: done

File: test/auth/google.test.js (hunk 6 above)
Comment (Adekoreday) [reply test/auth/google.test.js]: done

**Review by chialuka — DISMISSED** [review]

**Review by henryade — DISMISSED** [review]

**Review by dinobi — APPROVED** [review]

**Pull request merged**